src/battle-scene.ts

```diff
@@ -1591,7 +1598,7 @@ export default class BattleScene extends SceneBase {
         return 0;
       case Species.GIMMIGHOUL:
         // Chest form can only be found in Mysterious Chest Encounter, if this is a game mode with MEs
-        if (this.gameMode.hasMysteryEncounters) {
+        if (this.gameMode.hasMysteryEncounters && !isEggPhase) {
```

Comment:
```suggestion
        if (this.currentBattle?.battleType === BattleType.MYSTERY_ENCOUNTER && !isEggPhase) {
```
So that regular wild Gimmighoul are also properly randomized.